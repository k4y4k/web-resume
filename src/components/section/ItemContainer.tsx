import * as React from 'react'
import Title from './ItemTitle'
import Subtitle from './ItemSubtitle'
import Dates from './ItemDates'
import Details from './ItemDetails'

interface ItemContainerTypes {
  title?: string
  subtitle?: string
  fromDate?: string
  toDate?: string
  details?: string
}

const ItemContainer = ({
  title,
  subtitle,
  fromDate,
  toDate,
  details,
}: ItemContainerTypes): JSX.Element => (
  <>
    <Title title={title} />
    <Subtitle subtitle={subtitle} />
    <Dates from={fromDate} to={toDate} />
    <Details details={details} />
  </>
)

export default ItemContainer
