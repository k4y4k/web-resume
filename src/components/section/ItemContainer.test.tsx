import * as React from 'react'
import { render, screen } from '@testing-library/react'
import ItemContainer from './ItemContainer'
const data = {
  company: 'Company',
  position: 'President',
  startDate: '2013-01-01',
  endDate: '2014-01-01',
  summary: 'Description...',
  highlights: ['Started the company'],
  link: 'https://duckduckgo.com',
}

describe('<ItemContainer />', () => {
  describe('experience item', () => {
    test('experience item', () => {
      render(
        <ItemContainer
          fromDate={data.startDate}
          summary={data.summary}
          toDate={data.endDate}
          title={data.position}
          subtitle={data.company}
          link={data.link}
        />
      )

      const title = screen.queryByText('Company')
      expect(title).toBeInTheDocument()

      const subtitle = screen.getByTestId('byline')
      expect(subtitle).toBeInTheDocument()

      const dates = screen.queryByTestId('sectionItemDates')
      expect(dates).toBeInTheDocument()

      const details = screen.queryByTestId('sectionItemDetails')
      expect(details).toBeInTheDocument()

      expect(screen.getByTestId('sectionItemContainer')).toMatchSnapshot()
    })
  })

  describe('education item', () => {
    test('education item', () => {
      render(
        <ItemContainer
          institution='University'
          area='Software Development'
          studyType='Bachelor'
          fromDate='2011-01-01'
          toDate='2013-01-01'
          courses={['DB1101 - Basic SQL']}
        />
      )

      const title = screen.getByText('Bachelor of Software Development')
      expect(title).toBeInTheDocument()

      const byline = screen.getByTestId('byline')
      expect(byline).toHaveTextContent('University | 2011 - 2013')

      const list = screen.getByText('DB1101 - Basic SQL')
      expect(list).toBeInTheDocument()

      expect(screen.getByTestId('sectionItemContainer')).toMatchSnapshot()
    })

    test('no courses? get nothing', () => {
      render(
        <ItemContainer
          institution='University'
          area='Software Development'
          studyType='Bachelor'
          fromDate='2011-01-01'
          toDate='2013-01-01'
        />
      )

      expect(screen.getByTestId('sectionItemContainer')).toMatchSnapshot()
    })
  })

  describe('conditionally add the seperator between name and date', () => {
    test('date only', () => {
      render(
        <ItemContainer
          fromDate={data.startDate}
          summary={data.summary}
          toDate={data.endDate}
          title={data.position}
          subtitle={''}
          link={data.link}
        />
      )

      expect(screen.getByTestId('sectionItemContainer')).toMatchSnapshot()
    })
  })
})
