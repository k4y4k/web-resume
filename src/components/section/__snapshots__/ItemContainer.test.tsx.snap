// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemContainer /> education item education item 1`] = `
<div
  data-testid="sectionItemContainer"
>
  <h1
    data-testid="itemTitle"
  >
    Bachelor of Software Development
  </h1>
  <div
    data-testid="byline"
  >
    <span
      data-testid="itemSubtitle"
    >
      University
    </span>
    |
    <p
      data-testid="sectionItemDates"
    >
      2011
       - 
      2013
    </p>
  </div>
  <div
    data-testid="sectionItemDetails"
  >
    <ul>
      

      <li>
        DB1101 - Basic SQL
      </li>
      

    </ul>
  </div>
</div>
`;

exports[`<ItemContainer /> education item no courses? get nothing 1`] = `
<div
  data-testid="sectionItemContainer"
>
  <h1
    data-testid="itemTitle"
  >
    Bachelor of Software Development
  </h1>
  <div
    data-testid="byline"
  >
    <span
      data-testid="itemSubtitle"
    >
      University
    </span>
    |
    <p
      data-testid="sectionItemDates"
    >
      2011
       - 
      2013
    </p>
  </div>
</div>
`;

exports[`<ItemContainer /> experience item experience item 1`] = `
<div
  data-testid="sectionItemContainer"
>
  <h1
    data-testid="itemTitle"
  >
    President
  </h1>
  <div
    data-testid="byline"
  >
    <span
      data-testid="itemSubtitle"
    >
      Company
    </span>
    |
    <p
      data-testid="sectionItemDates"
    >
      2013
       - 
      2014
    </p>
  </div>
  <div
    data-testid="sectionItemDetails"
  >
    <p>
      Description...
    </p>
  </div>
</div>
`;
